### PR TITLE
docs: fix override conf note

### PR DESCRIPTION
**Pull request by zmstone**

### CODE REVIEW

File: scripts/merge-config.escript [hunk 1]
```diff
@@ -17,7 +17,7 @@ main(_) ->
                                case filelib:is_regular(CfgFile) of
                                    true ->
                                        {ok, Bin1} = file:read_file(CfgFile),
-                                       [Acc, io_lib:nl(), Bin1];
+                                       [Acc, "\n", Bin1];
```
Comment (qzhuyan) [inline scripts/merge-config.escript:20]: why this change? was the old could work on windows as well?

File: apps/emqx_conf/etc/emqx_conf.conf [hunk 2]
```diff
@@ -1,5 +1,11 @@
-## NOTE: The configurations in this file will be overridden by
-## `<path-to-emqx-installation>/data/emqx_overrides.conf`
+## NOTE:
+## Configs in this file might be overridden by:
+## 1. Environment variables start with 'EMQX_' prefix.
```
Comment (mononym) [inline apps/emqx_conf/etc/emqx_conf.conf:3]: nit: Insert 'which' or 'that' in front of 'start' so it reads 'Environment variables that start...'

File: apps/emqx_conf/etc/emqx_conf.conf [hunk 3]
```diff
@@ -1,5 +1,11 @@
...
+## 2. File {{ platform_data_dir }}/configs/cluster-override.conf
+## 3. File {{ platform_data_dir }}/configs/local-override.conf
+##
+## The *-override.conf files are overwritten at runtime when changes
+## are made from EMQ X dashboard UI, management HTTP API, or CLI
```
Comment (mononym) [inline apps/emqx_conf/etc/emqx_conf.conf:8]: nit: Missing period

File: apps/emqx_conf/etc/emqx_conf.md [hunk 4]
```diff
@@ -9,12 +9,15 @@ From bottom up:
...
-1. Local override: The path of which is configured by `local_override_conf_file`.<br>
-   Special overrides applied to local node.
+1. Cluster overrides: `${data_dir}/cluster-override.conf`
+1. Local node overrides: `${data_dir}/configs/local-override.conf`
 
-For detailed override rules, see [Config overlay rules](#config-overlay-rules).
+Where `${data_dir}` is configurable from `node.data_dir`.
+
+The `*-override.conf` files are overwritten at runtime when changes
+are made from dashboard UI, management HTTP API, or CLI
```
Comment (mononym) [inline apps/emqx_conf/etc/emqx_conf.md:18]: nit: Missing period

File: scripts/merge-config.escript (hunk 1 above)
Comment (zmstone) [reply scripts/merge-config.escript:20]: I thought `io_lib:nl()` returns `"\r\n"` for windows.
Now that I had a look at the source, it's actually `"\n"`

**Review by mononym — APPROVED** [review]
